src/Processors/QueryPlan/FilterStep.cpp

```diff
@@ -76,18 +184,45 @@ void FilterStep::transformPipeline(QueryPipelineBuilder & pipeline, const BuildQ
 void FilterStep::describeActions(FormatSettings & settings) const
 {
     String prefix(settings.offset, settings.indent_char);
+
+    auto cloned_dag = actions_dag.clone();
+
+    std::vector<ActionsAndName> and_atoms;
+    if (!actions_dag.hasStatefulFunctions())
+        and_atoms = splitAndChainIntoMultipleFilters(cloned_dag, filter_column_name);
```

Comment:
After building pipeline,  `action_dag`  is cleared. calling `describeActions` will failed at `splitAndChainIntoMultipleFilters`

Reply:
This is true for most other steps as well.
Afiak we really call explain after pipeline is build.
I was planning to make `explain` destructive as well; but it is not convenient while debugging.